output/eu_crude/analysis.py

```diff
@@ -74,8 +78,148 @@ def get_waste_id(resource_array):
     return set(wasteid)
 
 
+def get_colors():
+    """ Gets colors from a matplotlib dictionary
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    Colores: array
+        array of color hex codes
+    """
+
+    colores = []
+    for hex in matplotlib.colors.cnames.items():
+        colores.append(hex[1])
+    return colores
+
+
+def get_power(filename):
+    """ Gets capacity vs time for every country
+        in stacked bar chart.
+
+    Parameters
+    ----------
+    filename: file
+        cyclus output file used
+
+    Returns
+    -------
+    plot of net capapcity vs time
+
+    """
+
+    sim_time = int(cur.execute('select endtime from finish').fetchone()[0]) + 1
+    timestep = np.linspace(0, sim_time, num=sim_time + 1)
+    powercap = []
+    reactor_num = []
+    countries = []
+    # get power cap values
+
+    governments = cur.execute('sELECT prototype, agentid FROM agententry\
+                              WHERE spec = ":cycamore:DeployInst"').fetchall()
+
+    entry = cur.execute('sELECT power_cap, agententry.agentid, parentid, entertime\
+                        FROM agententry INNER JOIN\
+                        agentstate_cycamore_reactorinfo\
+                        ON agententry.agentid =\
+                        agentstate_cycamore_reactorinfo.agentid').fetchall()
+
+    exit = cur.execute('sELECT power_cap, agentexit.agentid, parentid, exittime\
+                        FROM agentexit INNER JOIN\
+                        agentstate_cycamore_reactorinfo\
+                        ON agentexit.agentid =\
+                        agentstate_cycamore_reactorinfo.agentid\
+                        INNER JOIN agententry\
+                        ON agentexit.agentid = agententry.agentid').fetchall()
+
+    # create empty array for each country_government
+    for gov in governments:
+        exec(gov[0] + '_power = []', globals())
+        exec(gov[0] + '_num = []', globals())
+        countries.append(gov[0])
+
+    # add and subtract capacity and num for every government
```

Comment:
so can this and the previous problem
be addressed with setting global variables maybe?